wire/bench_test.go

```diff
@@ -604,31 +604,36 @@ func BenchmarkTxHash(b *testing.B) {
 // BenchmarkDoubleHashB performs a benchmark on how long it takes to perform a
 // double hash returning a byte slice.
 func BenchmarkDoubleHashB(b *testing.B) {
-	var buf bytes.Buffer
-	if err := genesisCoinbaseTx.Serialize(&buf); err != nil {
-		b.Errorf("Serialize: unexpected error: %v", err)
-		return
-	}
-	txBytes := buf.Bytes()
-
-	b.ResetTimer()
 	for i := 0; i < b.N; i++ {
+		var buf bytes.Buffer
+		if err := genesisCoinbaseTx.Serialize(&buf); err != nil {
```

Comment:
This ends up including the serialization over head in each invocation, which I don't think is what we want? So you can either use `StopTimer`, or just move it back to lie _outside_ the loop.

Reply:
Re-reading the commit message, I think that' was the purpose of the `BenchmarkTxHash` benchmark: to account for that extra overhead. Also if you add `ReportAllocs`, then all these calls will also report on the allocations.